### DISCUSSION TITLE
Config Updates OTA?

### DISCUSSION THREAD

**Discussion by marcus-j-davies** [zwave-js/zwave-js#1289] [Ideas] [answered]
Thought i'd start it here - just in case it has no interest?

But would a method of say, `driver.downloadLatestDeviceConfigs()` have any interest with this project.

lets say, I have an app - that uses a version of zwave-js which is a few months old.
during this time, more device configurations, or device compat flags are created.

Wouldn't it be ideal for some user to call `driver.downloadLatestDeviceConfigs()`
which would pull the latest device configs/patches down, allowing the currently installed environment, to receive support for devices, that were not originally known.

This is strictly for device configurations, obviously  bug fixes and alike will still be pursued in the normal way.
but it could allow the config/devices folder, to receive updates over the course of a few versions.

Each release of the configurations folder could have a hash associated, and it is that hash that is checked.

Sorry, my professional role is to replace off-the-shelf with in-house systems, so have a tendency to get tourettes with ideas.
:open_mouth:

**Comment by AlCalzone** [comment, marked as answer]
Already working on it :)
==> https://github.com/zwave-js/node-zwave-js/pull/1271

Updating from the client side is the next step.

**Reply by marcus-j-davies** [reply]
Nice!

Just walk away Marcus, just walk away......
